bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Control.java

```diff
@@ -3563,6 +3563,12 @@ public void setMenu (Menu menu) {
 	this.menu = menu;
 }
 
+void setNativeZoomForGCData(GCData data) {
+	int nativeDeviceZoom = getShell().getNativeZoom();
+	data.nativeDeviceZoom = nativeDeviceZoom;
+	data.deviceZoom = DPIUtil.getZoomForAutoscaleProperty(nativeDeviceZoom);
+}
```

Comment:
This is a case of "[feature envy](https://refactoring.guru/smells/feature-envy)"

If you address [Heiko's comment](https://github.com/eclipse-platform/eclipse.platform.swt/pull/1214/files#r1599589995) and manage to remove the fields from `GCData` altogether then this method wouldn't be necessary. 

WDYT, would that be possible?